### DISCUSSION TITLE
No output for carbon flux

### DISCUSSION THREAD

**Discussion by gamlingdang** [NCAR/hrldas#154] [Q&A] [answered]
Hi, everyone.
I am a new user to Noah mp. According to the tutorial  (Note1_Single_Point.ipynb), I set up the model and ran the single-point case. In the output file named 199801020030.LDASOUT_DOMAIN1, the variables related to carbon flux (e.g. GPP, NEE, NPP) had no output or were missing. The other variables like temperature or LAI had detailed values. There is no error when running ./create_point_data.exe and ./hrldas.exe. All settings were the same as in the tutorial. Do I need to set some additional options if I want to get the result of carbon flux? Thanks.
![ncview1](https://github.com/NCAR/hrldas/assets/40694098/a43f3056-eaa2-4306-a3a1-148b5ce86bb8)

**Comment by cenlinhe** [comment, marked as answer]
You will need to use dynamic vegetation option = 2, 5, or 6. By default, it is 4.

DYNAMIC_VEG_OPTION          = 4   ! options for dynamic vegetation   [default = 4]
                                                             !   1 -> off (use table LAI; use FVEG = SHDFAC from input)
                                                             !   2 -> on  (together with OPT_CRS = 1)
                                                             !   3 -> off (use table LAI; calculate FVEG)
                                                             ! **4 -> off (use table LAI; use maximum vegetation fraction)
                                                             !   5 -> on  (use maximum vegetation fraction)
                                                             !   6 -> on  (use FVEG = SHDFAC from input)
                                                             !   7 -> off (use input LAI; use FVEG = SHDFAC from input)
                                                             !   8 -> off (use input LAI; calculate FVEG)
                                                             !   9 -> off (use input LAI; use maximum vegetation fraction)

**Reply by gamlingdang** [reply]
Yes. When setting dynamic vegetation option = 2, 5, or 6, there were the carbon flux variables. Thanks a lot, Dr. He.
